File: 2016/j5.cpp
#include <iostream>
#include <algorithm>
#include <functional>

using namespace std;

int main() {
    int q, n;
    scanf("%d", &q);
    scanf("%d", &n);
    int a[n], b[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    for (int i = 0; i < n; i++) {
        scanf("%d", &b[i]);
    }
    sort(a, a + n);
    if (q == 1) {
        sort(b, b + n);
    }
    else {
        sort(b, b + n, greater<int>());
    }
    int sum = 0;
    for (int i = 0; i < n; i++) {
        sum += max(a[i], b[i]);
    }
    printf("%d\n", sum);
    return 0;
}

File: 2016/j5.py
# I made a silly mistake in j5.cpp when debugging it...
# I wrote this to proof that the algorithm works...
# Both j5.py and j5.cpp work now.

q = int(input())
n = int(input())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

a.sort()
if q == 1:
    b.sort()
else:
    b.sort(reverse=True)

print(sum(max(x, y) for x, y in zip(a, b)))
